lib/srv/desktop/tdp/proto.go

```diff
@@ -1102,6 +1114,258 @@ func decodeSharedDirectoryListResponse(in peekReader) (SharedDirectoryListRespon
 	}, nil
 }
 
+// SharedDirectoryReadRequest is a message sent by the server to the client to request
+// bytes to be read from the file at the path and starting at byte offset.
+type SharedDirectoryReadRequest struct {
+	CompletionID uint32
+	DirectoryID  uint32
+	Path         string
+	PathLength   uint32
+	Offset       uint64
+	Length       uint32
+}
+
+func (s SharedDirectoryReadRequest) Encode() ([]byte, error) {
+	buf := new(bytes.Buffer)
+
+	buf.WriteByte(byte(TypeSharedDirectoryReadRequest))
+	binary.Write(buf, binary.BigEndian, s.CompletionID)
+	binary.Write(buf, binary.BigEndian, s.DirectoryID)
+	// binary.Write(buf, binary.BigEndian, s.PathLength)
```

Comment:
Why is this commented out?

Reply:
The commented code should be removed. At first I wrote path length + path string in two separate calls, but then found out that we have a helper function that does the same and I've changed it. In the next line you can see that I'm using `encodeString` function.